src/Bot.js

```diff
@@ -57,12 +50,20 @@ class Bot {
     reviveChannels(date) {
         this.channels.forEach(channel => {
             if (date.getTime() - channel.updated.getTime() > channel.delay * 1000) {
-                let source = this.chooseSourceFor(channel);
-                channel.callback(source ? source.getMessage() : '');
+                const source = this.chooseSourceFor(channel);
+                channel.callback(source ? source.getMessage() : undefined);
             }
         });
     }
 
+    /**
+     * Configure a new source
+     * @param {StaticSource} source
+     */
+    addSource(source) {
+        this.sources.push(source);
+    }
```

Comment:
Why move it?

Reply:
Grouping source-related methods together

Reply:
Hehe I was grouping configuration methods together